packages/flutter/lib/src/services/image_resource.dart

```diff
@@ -16,13 +16,14 @@ typedef void ImageListener(ui.Image image);
 /// image object might change over time, either because the image is animating
 /// or because the underlying image resource was mutated.
 class ImageResource {
```

Comment:
Ah, I understand now.

Here's one alternative:
```dart
   // (make sure the constructor doesn't have a default for scale)
  double get scale => _scale;
  double _scale;
  void set scale(double value) {
    assert(value != null);
    assert(_scale == null); // this is the key, it makes it only writable once
    assert(_image == null); // make sure you set the scale first, otherwise the listener will fire too early
    // alternatively, fire the listener only when both are set
    _scale = value;
  }
```